test/e2e/gateway_api_test.go

```diff
@@ -0,0 +1,189 @@
+//go:build e2e
+// +build e2e
+
+package e2e
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	configv1 "github.com/openshift/api/config/v1"
+	gwapi "github.com/openshift/cluster-ingress-operator/pkg/operator/controller/gatewayapi"
+
+	corev1 "k8s.io/api/core/v1"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"sigs.k8s.io/gateway-api/apis/v1beta1"
+)
+
+const (
+	testHostname = "*.example.com"
+)
+
+// Define here so we can delete them in the defer section
+var gateway v1beta1.Gateway
+var gatewayClass v1beta1.GatewayClass
```

Comment:
It isn't clear why you need to declare these here.  Note that [function literals in Go](https://go.dev/ref/spec#Function_literals) are [closures](https://en.wikipedia.org/wiki/Closure_(computer_programming)), so there shouldn't be a problem with using local variables in deferred funcs.  You can always have `testCrds` return a pointer, or give `testCrds` a pointer to the variable where you want it to store a value, so it isn't obvious why you cannot use function-local variables.

Reply:
Headsup - I changed alot to do this, so it's going to be a little extra to review.